src/main/java/slimeknights/tconstruct/smeltery/block/entity/CastingTankBlockEntity.java

```diff
@@ -0,0 +1,370 @@
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.item.BlockItem;
+import net.minecraft.world.item.BucketItem;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.phys.BlockHitResult;
+import net.minecraftforge.client.model.data.ModelData;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.ForgeCapabilities;
+import net.minecraftforge.common.util.LazyOptional;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.FluidType;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidHandlerItem;
+import net.minecraftforge.items.ItemHandlerHelper;
+import net.minecraftforge.items.wrapper.SidedInvWrapper;
+import slimeknights.mantle.fluid.FluidTransferHelper;
+import slimeknights.mantle.fluid.transfer.FluidContainerTransferManager;
+import slimeknights.mantle.fluid.transfer.IFluidContainerTransfer;
+import slimeknights.tconstruct.TConstruct;
+import slimeknights.tconstruct.common.network.InventorySlotSyncPacket;
+import slimeknights.tconstruct.common.network.TinkerNetwork;
+import slimeknights.tconstruct.library.client.model.ModelProperties;
+import slimeknights.tconstruct.library.fluid.FluidTankAnimated;
+import slimeknights.tconstruct.library.utils.NBTTags;
+import slimeknights.tconstruct.shared.block.entity.TableBlockEntity;
+import slimeknights.tconstruct.smeltery.TinkerSmeltery;
+import slimeknights.tconstruct.smeltery.item.TankItem;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class CastingTankBlockEntity extends TableBlockEntity implements ITankBlockEntity, WorldlyContainer {
+  /** Max capacity for the tank */
+  public static final int DEFAULT_CAPACITY = FluidType.BUCKET_VOLUME * 2;
+  // slots
+  public static final int INPUT = 0;
+  public static final int OUTPUT = 1;
+  private static final Component NAME = TConstruct.makeTranslation("gui", "casting");
+
+  /** Internal fluid tank instance */
+  @Getter
+  protected final FluidTankAnimated tank;
+  /** Capability holder for the tank */
+  private final LazyOptional<IFluidHandler> fluidHolder;
+  /** Last comparator strength to reduce block updates */
+  @Getter @Setter
+  private int lastStrength = -1;
+
+  /**
+   * Gets the capacity for the given block
+   * @param block  block
+   * @return  Capacity
+   */
+  public static int getCapacity(Block block) {
+    if (block instanceof ITankBlock) {
+      return ((ITankBlock) block).getCapacity();
+    }
+    return DEFAULT_CAPACITY;
+  }
```

Comment:
We probably don't need to do the whole variable capacity thing; it just exists for the tank variants like lantern and ingot tanks. For the casting tank I don't plan to have variants, so probably just hardcode to 4000mb for now. Can always abstract it later if needed, but better than adding in abstractions for unknown future plans.